bun/src/query/queryUtils.ts

```diff
@@ -3,66 +3,70 @@ import { ValueOf } from "type-fest";
 import type { TraverserMap, GroovyTraversal } from "groovy/dsl";
 
 /**
- * Generic return type for a gremlin traversal
- * currently supports {@link GroovyTraversal} for chaining and {@link TraverserMap} for results
+ * Generic return type for a {@link GroovyTraversal} invocation
+ * currently supports {@link GroovyTraversal} for chaining and {@link TraverserMap} for traversal results
  */
 export interface NextT {
```

Comment:
this shiz isnt scalable, typescript should automatically detect the return type
- try to overload the fn
- create distinct fns for each and return that shiz